lib/elastic_record/index/deferred.rb

```diff
@@ -57,15 +59,19 @@ def flush_deferred_actions!
             deferred_actions.clear
           end
 
-          def search_request(method, args)
+          def index_from_request(method, args)
             if method == :json_get && args.first =~ /^\/(.*)\/_m?search/
               $1.partition('/').first
             end
           end
 
           READ_METHODS = [:json_get, :head]
-          def read_request?(method, args)
-            READ_METHODS.include?(method) || (method == :json_post && args.first =~ /^\/_search\/scroll/)
+          def expedite_request?(method, args)
+            READ_METHODS.include?(method) || (method == :json_post && expedited_json_post?(args.first))
+          end
+
+          def expedited_json_post?(arg)
+            arg =~ /^\/_search\/scroll/ || arg =~ /^(.*)\/_pit/
```

Comment:
`expedited` implies that this request is cutting to the front of the queue (like when you order expedited processing with a passport application).

```suggestion
          def flush_deferred_actions?(method, args)
            READ_METHODS.include?(method) || flush_deferred_actions_for_json_post?(method, args.first)
          end
          
          POST_PATHS_THAT_REQUIRE_FLUSH = [
            /^\/_search\/scroll/
            /^(.*)\/_pit/
          ].freeze
          
          def flush_deferred_actions_for_json_post?(method, arg)
            method == :json_post && POST_PATHS_THAT_REQUIRE_FLUSH.any? { |path| arg =~ path } 
```